reference/client-access/python-sdk.md

```diff
@@ -0,0 +1,287 @@
+using the prefix character `:`. The values of the matching properties of `args`
+are bound to each placeholder.
+
+libSQL supports the same named placeholder prefix characters as SQLite: `:`,
+`@`, and `$`.
+
+## Execute a batch of statements
+
+:::info
+
+Be sure to read the [common section on batches] for libSQL clients to understand
+their behavior.
+
+:::
+
+To execute multiple statements in a transaction, use the `batch()` method on the
+client object, passing it an array of statements. The array may contain any type
+of statement that is also accepted by
+[`execute()`](#execute-a-single-statement). `batch()` an array of
+[ResultSet](#resultset) objects (one for each statement).
+
+The following code inserts a row for uid3 in two different tables using a
+transaction that commits them both at the same time.
+
+```py
+rss = client.batch([
+    libsql_client.Statement(
+        sql="insert into example_users values (?, ?)",
+        args=[ "uid3", "uid3@turso.tech" ]
+    ),
+    libsql_client.Statement(
+        sql="insert into example_scores values (?, ?, ?)",
+        args=[ "uid3", 1, 200 ]
+    ),
+])
+
+# rss[0].columns == ()
+# rss[0].rows == []
+# rss[0].rows_affected == 1
+# rss[1].columns == ()
+# rss[1].rows == []
+# rss[1].rows_affected == 1
+```
+
+## Interactive transactions
+
+:::info
+
+Be sure to read the [common section on interactive transactions] for libSQL
+clients to understand their behavior.
+
+:::
+
+Use the `transaction()` method on the client object to create a Transaction
+object to control the transaction. It provides the following methods:
+
+| Method | Description |
+| --- | --- |
+| `execute()` | Similar to the normal client `execute()`, except within the context of the transaction |
+| `commit()` | Commits all the write statements in the transaction |
+| `rollback()` | Rolls back the entire transaction |
+| `close()` | Immediately stops the transaction - must be called if the transaction was not committed or rolled back in order to free resources |
+
+The following code uses an interactive transaction to update a user’s level
+score, but only if it’s greater than the one that currently exists:
+
+```py
+uid = "uid1"
+level = 1
+new_score = 200
+
+transaction = client.transaction()
+rs = transaction.execute(
+    stmt="select score from example_scores where uid = ? and level = ?",
+    args=[ uid, level ]
+)
+
+# rs.columns == ('score',)
+# rs.rows[0] == 100
+
+old_score = rs.rows[0]["score"]
+if new_score > old_score:
+    transaction.execute(
+        stmt="update example_scores set score = ? where uid = ? and level = ?",
+        args=[ new_score, uid, level ]
+    )
+
+transaction.commit()
+```
+
+## ResultSet
+
+Database queries always yield a ResultSet object. This object has the following
+attribute:
+
+| Attribute | Type | Description |
+| --- | --- | --- |
+| `rows` | `List[Row]` | A list of Row objects containing the row values, empty for write operations |
```

Comment:
We may also mention that the `ResultSet` object behaves like a sequence of `Row`s: you can do `rs[1]`, `len(rs)` or `for row in rs`.